libraries/AP_Scripting/lua_bindings.cpp

```diff
@@ -802,6 +803,21 @@ int lua_serial_writestring(lua_State *L)
     return 1;
 }
 
+int lua_serial_read(lua_State *L) {
+    binding_argcheck(L, 1);
+
+    AP_Scripting_SerialAccess * port = check_AP_Scripting_SerialAccess(L, 1);
+
+    uint8_t c;
+    if (port->read(c)) {
+        lua_pushinteger(L, c);
+    } else {
+        lua_pushnil(L); // error, return nil
```

Comment:
Return 0

Reply:
We have to push nil specifically so that we return a value. Otherwise e.g. `string.char(port:read())` will return an empty string if there is no data available instead of raising an error that it tried to do something with nil. (Previously, it would raise an error that -1 is out of range).